float32[3] state
float32[3] state_dot